kafka-client/src/main/java/dev/responsive/kafka/internal/db/partitioning/TablePartitioner.java

```diff
@@ -31,27 +33,15 @@ public interface TablePartitioner<K, P> {
    */
   P metadataTablePartition(final int kafkaPartition);
 
-  static <K> TablePartitioner<K, Integer> defaultPartitioner() {
-    return new DefaultPartitioner<>();
-  }
-
   /**
-   * A default, no-op partitioner that is one-to-one with the Kafka partition
-   * and has only a single dimension to the partitioning key
-   *
-   * @param <K> the record key type
+   * @param key             the data key
+   * @param kafkaPartition  the partition in kafka
+   * @return whether {@code key} belongs in {@code kafkaPartition}
    */
-  class DefaultPartitioner<K> implements TablePartitioner<K, Integer> {
-
-    @Override
-    public Integer tablePartition(final int kafkaPartition, final K key) {
-      return kafkaPartition;
-    }
-
-    @Override
-    public Integer metadataTablePartition(final int kafkaPartition) {
-      return kafkaPartition;
-    }
+  boolean belongs(final Bytes key, final int kafkaPartition);
```

Comment:
seems cleaner to just have a method that returns the partition for a key, or the number of partitions so then we don't have a bunch of impls that throw unsupportedoperationexception than to have those impls understand the context of why it's being called and throw accordingly.

Reply:
discussed offline, the exceptions are less about the API and more a shortcut in order to avoid needing to pass in the numPartitions to all the other implementations. eventually it would make sense to implement this method in those as well.